src/_guides/language/type-system.md

```diff
@@ -466,26 +460,36 @@ DartPad: {{site.dartpad}}/4a2a9bc2242042ba5338533d091213c0
 [Try it in DartPad]({{site.dartpad}}/4a2a9bc2242042ba5338533d091213c0).
 {% endcomment %}
 
-What about going in the other direction? Can you assign an `Animal` list to a `List<Cat>`?
+What about going in the other direction? 
+Can you assign an `Animal` list to a `List<Cat>`?
 
 {:.passes-sa}
-<?code-excerpt "strong/lib/strong_analysis.dart (generic-type-assignment-Animal)" replace="/Animal/[!$&!]/g"?>
+<?code-excerpt "lib/strong_analysis.dart (generic-type-assignment-Animal)" replace="/Animal/[!$&!]/g"?>
 {% prettify dart tag=pre+code %}
 List<Cat> myCats = <[!Animal!]>[];
 {% endprettify %}
 
-This assignment passes static analysis,
-but it creates an implicit cast. It is equivalent to:
+This assignment does not pass static analysis 
+as it creates an implicit downcast, 
+which are disallowed from non-`dynamic` types such as `Animal`.
+
+{{site.alert.version-note}}
+  In packages that use a [language version][] before 2.12
+  (when support for [null safety][] was introduced),
+  code can implicitly downcast from these non-`dynamic` types.
+  You can disallow these non-`dynamic` downcasts
+  by specifying `implicit-casts: false` in the [analysis options file.][analysis]
+{{site.alert.end}}
 
-<?code-excerpt "strong/lib/strong_analysis.dart (generic-type-assignment-implied-cast)" replace="/as.*(?=;)/[!$&!]/g"?>
+For this code to pass static analysis, 
+you will instead need to use an explicit cast, 
+which may fail at runtime.
```

Comment:
```suggestion
To make this code pass static analysis, 
use an explicit cast, 
which might fail at runtime.
```

Tweaks to make it more direct and use [present tense](https://developers.google.com/style/tense), plus avoid the word [_may_](https://developers.google.com/style/word-list#may).